### PR TITLE
Giscus discussion support

### PR DESCRIPTION
**Pull request by lslezak**
- Allow discussions below the blog posts
- Uses the [Giscus](https://giscus.app/) application which internally uses the GitHub discussions

- Preview: https://agama-preview-pull-24.surge.sh/blog
- See testing discussion at https://agama-preview-pull-24.surge.sh/blog/2023/02/16/d-installer-0-7 which displays this GitHub discussion: https://github.com/agama-project/agama-project.github.io/discussions/20 (I'll delete my testing comments later :wink: )

### CODE REVIEW

File: src/theme/BlogPostItem/index.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      <BlogPostItem {...props} />
+      <hr />
+      <h2>Comments</h2>
+      <p>
+        We are interested in your feedback, feel free to leave a comment here!
+      </p>
+      <p>
+        To comment you need a <a href="https://github.com/">GitHub</a> account
+        and authorize the <a href="https://giscus.app/">Giscus</a> application.{" "}
+        {discussionLink}
```
[the remark this answers is not in the document]
Comment (lslezak) [reply src/theme/BlogPostItem/index.tsx:97]: Ggood point, in most blog posts we write contacts at the end, like https://agama-preview-pull-24.surge.sh/blog/2023/02/16/d-installer-0-7#conslusions. We could put that in the "Comments" section instead and avoid repeating that.

File: README.md [hunk 2]
```diff
@@ -72,3 +72,82 @@ meaningful. The blog post itself then can contain a longer title specified in th
...
+Advantages:
+
+- Faster loading, no need to run a GitHub query, the discussion URL is already known
+- Avoids using a wrong discussion because of the fuzzy GitHub search (see above)
+- The first user can add a comment directly to GitHub discussion without need to authorize the
+  Giscus application
+
+Disadvantages:
+
+- The discussion needs to be created manually in advance by the Agama developers, it cannot be
```
Comment (dgdavid) [inline README.md:141]: I wonder if there is no way to automate the discussion creation at the time to merge the PR. Some hook doing the work previously and injecting the id.

That way, the only disadvantage would be to have an empty discussion forever, not a big deal.

**Review by dgdavid — APPROVED** [review]
LGTM, just a question/crazy idea, although I'm 99% sure you already explored the topic.

File: src/theme/BlogPostItem/index.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  // register/unregister the Giscus message handler
+  useEffect(() => {
+    window.addEventListener("message", handleGiscusMessage);
+    return () => {
+      window.removeEventListener("message", handleGiscusMessage);
+    };
+  }, []);
+
+  // if there is an already existing discussion show a link to it
+  const discussionLink = discussionUrl ? (
```
Comment (dgdavid) [inline src/theme/BlogPostItem/index.tsx:68]: I think we can drop this and make the whole thing a bit short. Rationale: we are already stating that alternatively user can go to Agama GitHub project page

Comment (dgdavid) [reply src/theme/BlogPostItem/index.tsx:68]: I mean. It's great to help users as much as possible and I see this contains a direct link to the discussion. But fue that it is already a bunch of text and the user can comment right away... If she/he decided to go to the repo, I assume they know where to look for  :)

Comment (dgdavid) [reply src/theme/BlogPostItem/index.tsx:68]: Let me explain it with an example now that I'm on front of the laptop instead on a mobile device :)

In its current shape, you have 

 > We are interested in your feedback, feel free to leave a comment here!
> 
> To comment here you need a [GitHub](https://github.com/) account and authorize the [Giscus](https://giscus.app/) application. Or you can comment directly at [this GitHub discussion](https://github.com/agama-project/agama-project.github.io/discussions/20) without authorizing the Giscus application.
> 
> Alternatively you can reach us at the [#yast](https://web.libera.chat/#yast) channel at the [Libera.Chat](https://libera.chat/), [YaST Development](https://lists.opensuse.org/archives/list/yast-devel@lists.opensuse.org) mailing list or the [Agama project](https://github.com/agama-project/agama) at GitHub.

and my proposal is

> We are interested in your feedback, feel free to leave a comment here!
> 
> :information_source:  To comment here you need a [GitHub](https://github.com/) account and authorize the [Giscus](https://giscus.app/) application.
> 
> Alternatively you can reach us at the [#yast](https://web.libera.chat/#yast) channel at the [Libera.Chat](https://libera.chat/), [YaST Development](https://lists.opensuse.org/archives/list/yast-devel@lists.opensuse.org) mailing list or the [Agama project](https://github.com/agama-project/agama) at GitHub.

Where the "Agama project" could be the link to the already started discussion if it exists.

But now I wrote it, I realize discussions are in agama-project/agama-project and not in agama-project/agama. But you got the idea.

Comment (lslezak) [reply src/theme/BlogPostItem/index.tsx:68]: I think it is not a duplication. The discussion link point directly to the blog post discussion. The GitHub link is a generic contact, can be used for reporting issues, code contribution, etc...

I'd keep both.

File: README.md (hunk 2 above)
Comment (lslezak) [reply README.md:141]: We would need to create the discussions also for all old blog posts, it's quite difficult to find out what is a "new" or "old" post...

**Pull request merged**